src/units/unit.hpp

```diff
@@ -210,7 +210,7 @@ class unit : public std::enable_shared_from_this<unit>
 
 	unit_ptr clone() const
 	{
-		return unit_ptr(new unit(*this));
+		return unit_ptr(std::shared_ptr<unit>(new unit(*this)));
```

Comment:
`return unit_ptr(std::make_shared<unit>(*this))`?

Reply:
yes i should do this, the reason i added that line is that i temporarily, replaced unit_ptr with a wrapper for std::shared_ptr<unit> without the constructor that takes 1 unit* to detect all cases hen unit* is converted into unit_ptr